mewarpx/mewarpx/mwxrun.py

```diff
@@ -54,6 +55,22 @@ def init_grid(self, xmin, xmax, zmin, zmax, nx, nz, max_grid_size=4):
         self._set_geom_str()
         self._set_grid_params()
 
+    def init_diags(self):
+        if len(self.simulation.diagnostics) > 0:
+            diag_periods = []
+            for diag in self.simulation.diagnostics:
+                diag_periods.append(diag.period)
+
+            self.step_interval = max(diag_periods)
+            min_period = min(diag_periods)
```

Comment:
Just to add on to Peter's comment below, I don't think we need to keep track of the minimum period, I believe we just need the step interval to be a multiple of each diag period.